### DISCUSSION TITLE
Remap the window mode?

### DISCUSSION THREAD

**Discussion by vincevargadev** [helix-editor/helix#9694] [Q&A] [answered]
I'd like to remap opening the [window](https://docs.helix-editor.com/keymap.html#window-mode) mode from `Ctrl-w` to something else (for example `;` followed up with a `w`).

```toml
[keys.normal.";"]
w = "window_mode"
```

This gives an error that there is no command named `window_mode` which certainly looks correct according to the [source code](https://github.com/helix-editor/helix/blob/master/helix-term/src/commands.rs), too.

**Is there any way to remap the window mode?**

I don't like relying on the Ctrl key, it feels very "jetbrainsy", and I'd like to experiment with a different kind of keymapping that doesn't need that many shifts, controls, alts, etc...

**Comment by vincevargadev** [comment]
Oh, found it?

So the good news is that I overlooked another alternative for window mode, which is available through the space, so I can just type "space w" and the window mode opens, so my original goal of getting rid of the Ctrl can be achieved.

The bad news is that it looks like the window mode cannot be remapped? See screenshot.

![Screenshot 2024-02-21 at 18 27 17](https://github.com/helix-editor/helix/assets/9061239/78b09e22-97e7-40ed-afdb-6050574e1df0)

**Comment by the-mikedavis** [comment, marked as answer]
Window mode and other minor modes like space mode, goto mode, view mode, etc. are just groups of keybindings. There isn't a way to rebind all of the keys in a minor mode currently - you need to remap each of commands in the mode to a new key:

```toml
[keys.normal.";".w]
w = "rotate_view"
s = "hsplit"
v = "vsplit"
# ... etc. ...
```

**Comment by Thinato** [comment]
I tried to use

```
[keys.normal.";".w]
...
```

but it looks like its outdated, this is what worked for me:

```
[keys.normal."C-w"]
"%" = "vsplit"
"\"" = "hsplit"
...
```
